### PR TITLE
Document boolean attributes in JSX

### PR DESCRIPTION
**Pull request by glenjamin**

### CODE REVIEW

**Comment by camsong** [comment]
:+1: It really help, please also add `required` as well.http://www.w3.org/TR/html5/forms.html#the-required-attribute

File: docs/docs/02.1-jsx-in-depth.md [hunk 1]
```diff
@@ -176,6 +176,29 @@ var person = React.createElement(
 );
 ```
 
+### Boolean Attributes
+
+Omitting the value of an attribute causes JSX to treat it as `true`. To pass `false` an attribute expression must be used. This often comes up when using HTML form elements, for attributes like `disabled`, `checked` and `readonly`.
```
Comment (spicyj) [inline docs/docs/02.1-jsx-in-depth.md]: This should be `readOnly` for React.

**Comment by spicyj** [comment]
Let's also add `required` as @camsong says. Instead of listing the JS output, what do you think of just writing that `disabled` and `disabled={true}` are equivalent with an example?

**Comment by glenjamin** [comment]
I'm updated the example to be more relevant to the common case which people seem to ask about.

**Comment by zpao** [comment]
@spicyj happy with this now?

**Comment by spicyj** [comment]
Yup, this is fine now. Thank you @glenjamin!

**Pull request merged**